### PR TITLE
AddedTask9AndTask10

### PR DESCRIPTION
**Pull request by rayus007**
**Task 9:** Complete implementing the 4 and 5 methods for the DB connector following the TTD approach
4. get all object
5. delete object

**Task 10:**
- Complete implementing the client_manager following the TTD approach
- Create an abstract class called DBConnector with abstract methods
- Rename the current DBConnector where Redis logic is, to be DBConnectorRedis

### CODE REVIEW

File: Task_10/tests/unit/test_client_manager.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    return cli_manager
+
+
+def test_save_document_mock(mocker):
+    id_test = "123"
+    object_to_save = {"123": {"a": "a"}}
+    cli_manager = ClientManager()
+    mocker.patch("redis.Redis.set").return_value = True
+    mocker.patch("redis.Redis.get").return_value = json.dumps(object_to_save)
+    result_save = cli_manager.save_document(id_test, object_to_save)
```
Comment (favipcj) [inline Task_10/tests/unit/test_client_manager.py]: you should be saving a client object when using the clientManager

File: Task_10/tests/unit/test_client_manager.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+
+def test_get_by_id_mock(mocker):
+    id_test = "123"
+    object_to_save = {"123": {"a": "a"}}
+    mocker.patch("redis.Redis.set").return_value = True
+    mocker.patch("redis.Redis.get").return_value = json.dumps(id_test)
+    cli_manager = ClientManager()
+    cli_manager.save_document(id_test, object_to_save)
+    result = cli_manager.get_document(object_to_save)
```
Comment (favipcj) [inline Task_10/tests/unit/test_client_manager.py]: same here, you should be saving and retrieving client objects please correct the other tests in this file too

File: Task_10/truck_delivery_rayus007/client_manager.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+
+class ClientManager(ContentManager):
+    """ Class representing a ClientManager """
+    def __init__(self):
+        super().__init__()
+        self.connector = DbConnectorRedis()
+
+    def save_document(self, key, document):
+        result_save = self.connector.save(f"{UID}_{CLIENT}_{key}", json.dumps(document))
```
Comment (favipcj) [inline Task_10/truck_delivery_rayus007/client_manager.py:14]: you should be receiving a client Object here and saving that

**Review by favipcj — CHANGES_REQUESTED** [review]

File: Task_10/truck_delivery_rayus007/db_connector_redis.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
+import redis
+import json
+from truck_delivery_rayus007.db_connector import DBConnector
+
+
+class DbConnectorRedis(DBConnector):
```
Comment (favipcj) [inline Task_10/truck_delivery_rayus007/db_connector_redis.py:6]: For implementing this class as a singleton class, please take a look at this https://python-patterns.guide/gang-of-four/singleton/#a-more-pythonic-implementation

File: Task_10/tests/unit/test_client_manager.py (hunk 2 above)
Comment (rayus007) [reply Task_10/tests/unit/test_client_manager.py]: Corrected.

File: Task_10/truck_delivery_rayus007/client_manager.py (hunk 3 above)
Comment (rayus007) [reply Task_10/truck_delivery_rayus007/client_manager.py:14]: I am getting/saving a Client object now

**Review by favipcj — APPROVED** [review]

**Pull request merged**